src/test/shared/utilities/timeoutUtils.test.ts

```diff
@@ -0,0 +1,54 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import * as timeoutUtils from '../../../shared/utilities/timeoutUtils'
+
+describe ('timeoutUtils', async () => {
+
+    describe ('Timeout', async () => {
+
+        it ('returns > 0 if the timer is still active', async () => {
+            const longTimer = new timeoutUtils.Timeout(300)
+            assert.strictEqual(longTimer.remainingTime > 0, true)
+            // bleed the timeout to not mess with other tests
+            await longTimer.timer
+        })
+
+        it ('returns 0 if timer is expired', async () => {
+            const shortTimer = new timeoutUtils.Timeout(1)
```

Comment:
Instead of waiting 100, could you do
await shortTimer.timer.then(() => assert test here)

Reply:
✔️

Reply:
This is flaky when waiting for the timer to run its course (it sometimes returns 1). I'm opting for a 10ms timeout just to be sure.